plugins/circleci/src/circleci-config.ts

```diff
@@ -226,6 +227,28 @@ const customOptionsOverlap = (
   })
 }
 
+const workflowJobOptionsOverlap = (
+  installation?: CircleCiWorkflowJob[],
+  other?: CircleCiWorkflowJob[]
+): boolean => {
+  if (!installation || !other) {
+    return false
+  }
+  return installation.some((installationWorkflowJob) => {
+    const otherWorkflowJob = other.find(({ name }) => installationWorkflowJob.name === name)
+    return (
+      otherWorkflowJob &&
+      ((installationWorkflowJob.runOnRelease !== undefined &&
+        otherWorkflowJob.runOnRelease !== undefined &&
+        installationWorkflowJob.runOnRelease !== otherWorkflowJob.runOnRelease) ||
+        (installationWorkflowJob.splitIntoMatrix !== undefined &&
+          otherWorkflowJob.splitIntoMatrix !== undefined &&
+          installationWorkflowJob.splitIntoMatrix !== otherWorkflowJob.splitIntoMatrix) ||
+        customOptionsOverlap(installationWorkflowJob.custom, otherWorkflowJob.custom))
```

Comment:
**nitpick**: don't really know how you could simplify this but it is _very_ dense right now! logic looks fine though?

Reply:
yeaaah i should probably refactor this logic into a function like `areDefinedAndEqual` shouldn't i